src/hotspot/share/gc/g1/g1CommittedRegionMap.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_GC_G1_G1COMMITTEDREGIONMAP_HPP
+#define SHARE_GC_G1_G1COMMITTEDREGIONMAP_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/bitMap.hpp"
+#include "utilities/macros.hpp"
+
+// Helper class to define a range [start, end) of regions.
+class HeapRegionRange : public StackObj {
+  // Inclusive start of the range.
+  uint _start;
+  // Exclusive end of the range.
+  uint _end;
+ public:
+  HeapRegionRange(uint start, uint end);
+
+  uint start() const { return _start; }
+  uint end() const { return _end; }
+  uint length() const { return _end - _start; }
+};
+
+class G1CommittedRegionMap : public CHeapObj<mtGC> {
```

Comment:
It would be nice to have a diagram of the region states and their transitions here.